### PR TITLE
feat(ipfs-manager): adapt RPC usage with POST methods

### PR DESCRIPTION
**Pull request by benjlevesque**
modern versions of IPFS now require using POST for every calls

https://github.com/ipfs/go-ipfs/pull/7097
https://github.com/ipfs/js-ipfs/issues/2971

Also in this PR: replace `axios-retry` since it seems to not retry POST requests

### CODE REVIEW

File: packages/ethereum-storage/test/ipfs-manager.test.ts [hunk 1]
```diff
@@ -76,24 +76,24 @@ describe('Ipfs manager', () => {
   it('allows to pin one file ipfs', async () => {
     await ipfsManager.add(content);
     const pinnedHash = await ipfsManager.pin([hash]);
-    expect(hash).toBe(pinnedHash[0]);
+    expect(pinnedHash[0]).toBe(hash);
```
Comment (benjlevesque) [inline packages/ethereum-storage/test/ipfs-manager.test.ts:79]: wrong order for expect

File: packages/ethereum-storage/test/ipfs-manager.test.ts [hunk 2]
```diff
@@ -146,10 +146,10 @@ describe('Ipfs manager', () => {
     const axiosInstanceMock = new MockAdapter(axiosInstance);
     axiosInstanceMock.onAny().networkError();
     await expect(ipfsManager.read(hash)).rejects.toThrowError('Network Error');
-    expect(axiosInstanceMock.history.get.length).toBe(retryTestErrorHandling.maxRetries + 1);
+    expect(axiosInstanceMock.history.post.length).toBe(retryTestErrorHandling.maxRetries + 1);
   });
 
-  it('timeout errors should not generate any retry', async () => {
+  it('timeout errors should generate retry', async () => {
```
Comment (benjlevesque) [inline packages/ethereum-storage/test/ipfs-manager.test.ts:152]: IMPORTANT. We now want to retry on timeout

**Review by alexandre-abrioux — APPROVED** [review]

File: packages/ethereum-storage/src/config.ts [hunk 3]
```diff
@@ -3,9 +3,9 @@ import { BigNumber } from 'ethers';
 
 // This contains default values used to use Ethereum Network and IPFS
 // if information are not provided by the user
-const config: any = {
+const config = {
```
Comment (benjlevesque) [inline packages/ethereum-storage/src/config.ts:6]: unrelated, annying any

File: packages/ethereum-storage/src/config.ts [hunk 4]
```diff
@@ -22,22 +22,19 @@ const config: any = {
...
-    expectedBootstrapNodes: [
-      '/dns4/ipfs-bootstrap.request.network/tcp/4001/ipfs/QmaSrBXFBaupfeGMTuigswtKtsthbVaSonurjTV967Fdxx',
-
-      '/dns4/ipfs-bootstrap-2.request.network/tcp/4001/ipfs/QmYdcSoVNU1axgSnkRAyHtwsKiSvFHXeVvRonGCAV9LVEj',
-
-      '/dns4/ipfs-2.request.network/tcp/4001/ipfs/QmPBPgTDVjveRu6KjGVMYixkCSgGtVyV8aUe6wGQeLZFVd',
-
-      '/dns4/ipfs-survival.request.network/tcp/4001/ipfs/Qmb6a5DH45k8JwLdLVZUhRhv1rnANpsbXjtsH41esGhNCh',
-    ],
+    expectedBootstrapNodes: {
```
Comment (benjlevesque) [inline packages/ethereum-storage/src/config.ts:32]: the format of the address changed, this accounts for the change while supporting older format

**Review by kevindavee — APPROVED** [review]

**Review by bertux — APPROVED** [review]

File: packages/ethereum-storage/src/ipfs-storage.ts [hunk 5]
```diff
@@ -139,4 +134,13 @@ export class IpfsStorage implements StorageTypes.IIpfsStorage {
...
     }
   }
+
+  static hasRequiredBootstrapNodes(actualList: string[]): boolean {
+    const expectedList = getIpfsExpectedBootstrapNodes();
+    const bootstrapNodeFoundCount: number = expectedList.filter((nodeExpected) =>
+      actualList.find((actual) => nodeExpected.test(actual)),
+    ).length;
+
+    return bootstrapNodeFoundCount === expectedList.length;
```
Comment (alexandre-abrioux) [inline packages/ethereum-storage/src/ipfs-storage.ts:144]:
```suggestion
    return expectedList.every((nodeExpected) => 
        actualList.some((actual) => nodeExpected.test(actual))
    )
```

**Review by alexandre-abrioux — APPROVED** [review]

**Pull request merged**